revup/config.py

```diff
@@ -4,10 +4,87 @@
 import logging
 import os
 import re
+from argparse import _StoreAction, _StoreFalseAction, _StoreTrueAction
+from typing import Any, Dict, List, Optional
 
 from revup.types import RevupUsageException
 
 
+class RevupArgParser(argparse.ArgumentParser):
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+
+    def add_argument(self, *args: Any, **kwargs: Any) -> argparse.Action:
+        """
+        For each boolean store_true action, add a corresponding "no" store_false action
+        with the same target.
+        """
+        action = super().add_argument(*args, **kwargs)
+
+        if isinstance(action, _StoreTrueAction):
+            no_options = []
+            for option_string in action.option_strings:
+                if option_string.startswith("--"):
+                    no_options.append("--no-" + option_string[2:])
+                elif option_string.startswith("-"):
+                    no_options.append("-n" + option_string[1:])
```

Comment:
Is this idiomatic?  I tend to think short cli args can be stacked like `tool -abcdef`, and yeah that's how argparse works also: https://docs.python.org/3/library/argparse.html#option-value-syntax

It seems like it'd be confusing to pass e.g. `-nx` and not know if that's short for `-n -x` or `--no-xlong-argument`, I think the way revup is set up here it'd always be `--no-xlong-argument` but I'm not sure if I've seen a cli tool where that'd be the case

Reply:
yeah true, fwiw we don't have any flags that use -n for roughly this reason (except stragglers in toolkit that can be changed) would this be better if we formalized that?

we can also get rid of the short syntax and use only --no- longform syntax, i'm not that attached to it

Reply:
I think I'd rather use only `--no-` longform

Reply:
sg i'll make that a new pr though, its not strictly related to this

Reply:
SGTM